src/crimson/osd/object_context_loader.cc

```diff
@@ -37,33 +37,38 @@ using crimson::common::local_conf;
   template<RWState::State State>
   ObjectContextLoader::load_obc_iertr::future<>
   ObjectContextLoader::with_clone_obc(const hobject_t& oid,
-                                      with_obc_func_t&& func)
+                                      with_obc_func_t&& func,
+                                      bool resolve_clone)
   {
     LOG_PREFIX(ObjectContextLoader::with_clone_obc);
     assert(!oid.is_head());
     return with_head_obc<RWState::RWREAD>(
       oid.get_head(),
-      [FNAME, oid, func=std::move(func), this](auto head, auto) mutable
-      -> load_obc_iertr::future<> {
+      [FNAME, oid, func=std::move(func), resolve_clone, this]
+      (auto head, auto) mutable -> load_obc_iertr::future<> {
       if (!head->obs.exists) {
 	ERRORDPP("head doesn't exist for object {}", dpp, head->obs.oi.soid);
         return load_obc_iertr::future<>{
           crimson::ct_error::enoent::make()
         };
       }
-      auto resolved_oid = resolve_oid(head->get_head_ss(), oid);
-      if (!resolved_oid) {
-        ERRORDPP("clone {} not found", dpp, oid);
-        return load_obc_iertr::future<>{
-          crimson::ct_error::enoent::make()
-        };
-      }
-      if (resolved_oid->is_head()) {
-        // See resolve_oid
-        return std::move(func)(head, head);
+      hobject_t clone_oid = oid;
+      if (resolve_clone) {
```

Comment:
if we don't do `resolve_oid`, chances are that the later `with_clone_obc_only` invocation will failed to find a proper `clone_obc`, so we need to handle `enoent` properly, am I right?

Reply:
>if we don't do resolve_oid, chances are that the later with_clone_obc_only invocation will failed to find a proper clone_obc, so we need to handle enoent properly, am I right?

The behavior here is not changed. Before the changes here, when trimming, we used with_clone_obc_direct which **didn't** resolve_oid and used it directly.